core/src/rpc/rpc_context.rs

```diff
@@ -481,14 +483,54 @@ pub trait RpcContext {
 	}
 
 	// ------------------------------
-	// Methods for querying
+	// Methods for relating
 	// ------------------------------
 
 	async fn relate(&self, _params: Array) -> Result<impl Into<Data>, RpcError> {
 		let out: Result<Value, RpcError> = Err(RpcError::MethodNotFound);
 		out
 	}
 
+	// ------------------------------
+	// Methods for running functions
+	// ------------------------------
+
+	async fn run(&self, params: Array) -> Result<impl Into<Data>, RpcError> {
+		let Ok((Value::Strand(Strand(func_name)), version, args)) = params.needs_three() else {
```

Comment:
Shouldn't we use `needs_two_or_three` here? Possibly even one two or three? You might want to invoke a function without parameters, and possibly you want to invoke a function without parameters that has no version, thus none

Reply:
a name and a version, the order is fixed. We can document that. I was just thinking it could safe a few bytes from the request haha. Microseconds!!!

Reply:
so do you think it should be either [name, version, params] or [name, params]

Reply:
I think it should accept these
```
[name]
[name, version]
[name, version, params]
```

Reply:
ok, that's reasonable